packages/server/src/sdk/app/rows/search/external.ts

```diff
@@ -133,32 +133,32 @@ export async function exportRows(
 
   let result = await search({ tableId, query: requestQuery, sort, sortOrder })
   let rows: Row[] = []
+  let headers
 
-  // Filter data to only specified columns if required
+  if (!tableName) {
+    throw new HTTPError("Could not find table name.", 400)
+  }
+  const schema = datasource.entities[tableName].schema
 
+  // Filter data to only specified columns if required
   if (columns && columns.length) {
     for (let i = 0; i < result.rows.length; i++) {
       rows[i] = {}
       for (let column of columns) {
         rows[i][column] = result.rows[i][column]
       }
     }
+    headers = Object.keys(schema).filter(header => columns.includes(header))
```

Comment:
The "columns" should never be bigger than the schema. Could we iterate through "columns" checking that is a valid schema key?

Reply:
Realised that this was completely unnecessary, I can just directly use the columns in this case, no need to do the filter.

Unit test added as well